lib/decimal.ex

```diff
@@ -814,6 +814,24 @@ defmodule Decimal do
   end
 
   @doc """
+  Returns the decimal represented as an integer. Fails when loss of precision
+  will occur.
+  """
+  def to_integer(%Decimal{sign: sign, coef: coef, exp: 0}), do: sign * coef
+
+  def to_integer(%Decimal{sign: sign, coef: coef, exp: exp}) when (exp > 0) do
+    to_integer(%Decimal{sign: sign, coef: coef * 10, exp: exp - 1})
+  end
+
+  def to_integer(%Decimal{sign: sign, coef: coef, exp: exp}) when (exp < 0) and (Kernel.rem(coef, 10) == 0) do
+    to_integer(%Decimal{sign: sign, coef: trunc(coef / 10), exp: exp + 1})
+  end
+
+  def to_integer(%Decimal{} = num) do
+    error(:invalid_operation, "not an integer", num)
```

Comment:
This should raise ArgumentError or FunctionClauseError because it's not a decimal arithmetic operation.